File: bestia-models/src/main/java/net/bestia/model/dao/BestiaDAO.java
package net.bestia.model.dao;

import org.springframework.data.repository.CrudRepository;

import net.bestia.model.domain.Bestia;

public interface BestiaDAO extends CrudRepository<Bestia, Integer> {

	/**
	 * Finds a bestia by its database name.
	 * 
	 * @param databaseName
	 *            The bestia database name.
	 * @return The found {@link Bestia} or NULL.
	 */
	public Bestia findByDatabaseName(String databaseName);

}

File: bestia-models/src/main/java/net/bestia/model/dao/BestiaDAO.kt
package net.bestia.model.dao

import org.springframework.stereotype.Repository

import net.bestia.model.domain.Bestia
import org.springframework.data.jpa.repository.JpaRepository

@Repository("bestiaDao")
interface BestiaDAO : JpaRepository<Bestia, Int> {

  /**
   * Finds a bestia by its database name.
   *
   * @param databaseName
   * The bestia database name.
   * @return The found [Bestia] or NULL.
   */
  fun findByDatabaseName(databaseName: String): Bestia?

}
